lib/source/defender/aws_iot_defender_mqtt.c

```diff
@@ -139,35 +153,100 @@ IotMqttError_t AwsIotDefenderInternal_MqttSubscribe( IotMqttCallbackInfo_t accep
     subscriptions[ 1 ].topicFilterLength = ( uint16_t ) strlen( _pRejectTopic );
     subscriptions[ 1 ].callback = rejectCallback;
 
-    return IotMqtt_SubscribeSync( _mqttConnection,
-                                  subscriptions,
-                                  2,
-                                  0,
-                                  _defenderToMilliseconds( AWS_IOT_DEFENDER_MQTT_SUBSCRIBE_TIMEOUT_SECONDS ) );
+    /* Call the asynchronous SUBSCRIBE function. */
+    status = IotMqtt_SubscribeAsync( _mqttConnection,
```

Comment:
Consider using the "+" wildcard so you don't have to make 2 subscriptions.  This halves the memory usage for subscriptions.  You will need to parse the topic to find what the wildcard represents, but there's already a function in aws_iot.h to do this.

There's also a function already in aws_iot.h that handles subscriptions.